synapse/storage/client_ips.py

```diff
@@ -93,6 +95,25 @@ def insert_client_ip(self, user_id, access_token, ip, user_agent, device_id,
 
         self._batch_row_update[key] = (user_agent, device_id, now)
 
+    @defer.inlineCallbacks
+    def _populate_monthly_active_users(self, user_id):
+        """Checks on the state of monthly active user limits and optionally
+        add the user to the monthly active tables
+
+        Args:
+            user_id(str): the user_id to query
+        """
+
+        store = self.hs.get_datastore()
+        if self.hs.config.limit_usage_by_mau:
+            is_user_monthly_active = yield store.is_user_monthly_active(user_id)
+            if is_user_monthly_active:
+                yield store.upsert_monthly_active_user(user_id)
```

Comment:
I would avoid having to do an update on every request. One way of avoiding this is to have the `is_user_monthly_active` return the timestamp and use that to only insert if the difference is greater than, say, a couple of hours. Since that DB function is cached we avoid doing queries most of the time.

Reply:
makes sense